test/runtime/CMakeLists.txt

```diff
@@ -0,0 +1,24 @@
+function(halide_define_runtime_internal_test NAME)
+    add_executable(runtime_internal_${NAME} ${NAME}.cpp)
+    target_link_libraries(runtime_internal_${NAME} PRIVATE Halide::Test)
+    target_include_directories(runtime_internal_${NAME} PRIVATE "${Halide_SOURCE_DIR}/src")
+
+    target_link_libraries(runtime_internal_${NAME} PRIVATE Halide::Runtime)
+    set_target_properties(runtime_internal_${NAME} PROPERTIES ENABLE_EXPORTS TRUE)
+    target_compile_options(runtime_internal_${NAME} PUBLIC -Wno-builtin-declaration-mismatch )
+    target_compile_definitions(runtime_internal_${NAME} PUBLIC HALIDE_VERSION=${Halide_VERSION}  )
+    target_compile_definitions(runtime_internal_${NAME} PUBLIC HALIDE_VERSION_MAJOR=${Halide_VERSION_MAJOR} )
+    target_compile_definitions(runtime_internal_${NAME} PUBLIC HALIDE_VERSION_MINOR=${Halide_VERSION_MINOR} )
+    target_compile_definitions(runtime_internal_${NAME} PUBLIC HALIDE_VERSION_PATCH=${Halide_VERSION_MINOR} )
```

Comment:
```suggestion
    target_compile_definitions(runtime_internal_${NAME} PUBLIC HALIDE_VERSION_PATCH=${Halide_VERSION_PATCH} )
```

Reply:
Also, you can place multiple definitions in `target_compile_definitions`:
```
target_compile_definitions(
    runtime_internal_${NAME}
    PRIVATE
    HALIDE_VERSION=${Halide_VERSION}
    HALIDE_VERSION_MAJOR=${Halide_VERSION_MAJOR}
    HALIDE_VERSION_MINOR=${Halide_VERSION_MINOR}
    HALIDE_VERSION_PATCH=${Halide_VERSION_PATCH}
    COMPILING_HALIDE_RUNTIME
    COMPILING_HALIDE_RUNTIME_TESTS
)
```
Also, the visibility here should be `PRIVATE`. Since there are executables, it's just a matter of style, but setting them to `PUBLIC` is kind of a head-scratcher.

Reply:
Ah, good catch! Yeah there's no reason they should be PUBLIC ... I'll set everything to PRIVATE.